DSharpPlus.UnifiedCommands/Message/Internals/MessageHandler.cs

```diff
@@ -0,0 +1,188 @@
+            case MessageResultType.FollowUp:
+                if (_newMessage is not null)
+                {
+                    result.Builder.WithReply(_newMessage.Id);
+                }
+
+                await _module.Message.Channel.SendMessageAsync(result.Builder);
+                break;
+            case MessageResultType.Edit:
+                if (_newMessage is not null)
+                {
+                    await _newMessage.ModifyAsync(result.Builder);
+                }
+
+                break;
+            default:
+                throw new ArgumentOutOfRangeException();
+        }
+    }
+
+    internal async Task BuildModuleAndExecuteCommandAsync()
+    {
+        try
+        {
+            long startTime = Stopwatch.GetTimestamp();
+
+            MessageModuleData moduleData = _data.Module;
+
+            _module = (MessageModule)moduleData.Factory.Invoke(_scope.ServiceProvider, null);
+            _module.Message = _message;
+            _module._handler = this;
+            _module.Client = _client;
+
+            object?[]? parameters = null;
+            if (_data.Parameters.Count != 0)
+            {
+                MessageConvertValues conversion = new(_values, _data, _message, _client);
+                try
+                {
+                    parameters = await conversion.StartConversionAsync();
+                }
+                catch (Exceptions.ConversionFailedException e)
+                {
+                    await _scope.ServiceProvider.GetRequiredService<IErrorHandler>().HandleConversionAsync(
+                        new InvalidMessageConversionError
+                        {
+                            Name = e.Name,
+                            IsPositionalArgument = e.IsPositionalArgument,
+                            Value = e.Value,
+                            Type = e.Type,
+                        }
+                        , _message);
+                    return;
+                }
+            }
+
+            if (_conditionBuilders.Count != 0)
+            {
+                MessageConditionHandler conditionHandler =
+                    new(_conditionBuilders);
+                bool shouldContinue =
+                    await conditionHandler.StartGoingThroughConditionsAsync(
+                        new MessageContext { Message = _message, Data = new(_name, _data.Method), Client = _client },
+                        _scope);
+                if (!shouldContinue)
+                {
+                    return;
+                }
+            }
+
+            TimeSpan elapsedTime = Stopwatch.GetElapsedTime(startTime);
+            _client.Logger.LogDebug("Took {NsExecution}ns to process everything", elapsedTime.TotalNanoseconds);
+
+            try
+            {
+                if (_data.ReturnsNothing)
+                {
+                    if (_data.IsAsync)
+                    {
+                        await (Task)_data.Method.Invoke(_module, BindingFlags.OptionalParamBinding,
```

Comment:
we should properly support ValueTask, and that includes not casting to a Task, which negates the benefits